src/client/serialize/daos_serialize.c

```diff
@@ -1175,3 +1244,1818 @@ daos_cont_deserialize_attrs(char *filename, uint64_t *num_attrs,
+		memset(large_key, 0, ENUM_LARGE_KEY_BUF);
+		akey_number = ENUM_DESC_NR;
+
+		akey_sgl.sg_nr     = 1;
+		akey_sgl.sg_nr_out = 0;
+		akey_sgl.sg_iovs   = &akey_iov;
+
+		key_buf = small_key;
+		key_buf_len = ENUM_DESC_BUF;
+		d_iov_set(&akey_iov, key_buf, key_buf_len);
+
+		/* get akeys */
+		rc = daos_obj_list_akey(*oh, DAOS_TX_NONE, &diov, &akey_number, akey_kds,
+					&akey_sgl, &akey_anchor, NULL);
+		if (rc == -DER_KEY2BIG) {
+			/* call list dkey again with bigger buffer */
+			key_buf = large_key;
+			key_buf_len = ENUM_LARGE_KEY_BUF;
+			d_iov_set(&akey_iov, key_buf, key_buf_len);
+			rc = daos_obj_list_akey(*oh, DAOS_TX_NONE, &diov, &akey_number,
+						akey_kds, &akey_sgl, &akey_anchor, NULL);
+			if (rc != 0) {
+				D_ERROR("Failed to list akeys "DF_RC"\n", DP_RC(rc));
+				D_GOTO(out, rc);
+			}
+		}
+
+		/* if no akeys returned move on */
+		if (akey_number == 0)
+			continue;
+
+		size = (akey_number + *total_akeys) * sizeof(struct akey);
+		*hdf5->ak = realloc(*hdf5->ak, size);
+		if (*hdf5->ak == NULL)
+			D_GOTO(out, rc =  -DER_NOMEM);
+
+		/* parse out individual akeys based on key length and number of dkeys returned */
+		for (akey_ptr = key_buf, j = 0; j < akey_number; j++) {
+			D_ALLOC(akey, key_buf_len);
+			if (akey == NULL)
+				D_GOTO(out, rc = -DER_NOMEM);
+			memcpy(akey, akey_ptr, akey_kds[j].kd_key_len);
+			d_iov_set(&aiov, (void*)akey, akey_kds[j].kd_key_len);
+			akey_val = &(*hdf5->ak)[*akey_index].akey_val;
+			D_ALLOC(akey_val->p, (uint64_t)akey_kds[j].kd_key_len * sizeof(char));
+			if (akey_val->p == NULL)
+				D_GOTO(out, rc = -DER_NOMEM);
+			memcpy(akey_val->p, (void*)akey_ptr, (uint64_t)akey_kds[j].kd_key_len);
+			akey_val->len = (uint64_t)akey_kds[j].kd_key_len; 
+			(*hdf5->ak)[*akey_index].rec_dset_id = *akey_index;
+
+			/* set iod values */
+			iod.iod_nr   = 1;
+			iod.iod_type = DAOS_IOD_SINGLE;
+			iod.iod_size = DAOS_REC_ANY;
+			iod.iod_recxs = NULL;
+			iod.iod_name  = aiov;
+
+			/* do a fetch (with NULL sgl) of single value type, and if that returns
+			 * iod_size == 0, then a single value does not exist. */
+			rc = daos_obj_fetch(*oh, DAOS_TX_NONE, 0, &diov, 1, &iod, NULL, NULL, NULL);
+			if (rc != 0) {
+				D_ERROR("Failed to fetch object "DF_RC"\n", DP_RC(rc));
+				D_GOTO(out, rc);
+			}
+
+			/* if iod_size == 0 then this is a DAOS_IOD_ARRAY type */
+			if ((int)iod.iod_size == 0) {
+				/* set single value field to NULL, 0 for array types */
+				single_val = &(*hdf5->ak)[*akey_index].rec_single_val;
+				single_val->p = NULL;
+				single_val->len = 0;
+
+				/* create a record dset only for array types */
+				memset(&rec_name, 0, rec_name_len);
+				path_len = snprintf(rec_name, rec_name_len, "%d", *akey_index);
+				if (path_len > rec_name_len) {
+					D_ERROR("Record name too long "DF_RC"\n", DP_RC(rc));
+					D_GOTO(out, rc);
+				}
+				rc = serialize_recx_array(hdf5, &diov, &aiov, rec_name, akey_index,
+				     			  oh, &iod, bytes_read);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs